mex.go

```diff
@@ -199,6 +208,7 @@ func (mexset *messageExchangeSet) newExchange(ctx context.Context, framePool Fra
 		msgID:     msgID,
 		ctx:       ctx,
 		recvCh:    make(chan *Frame, bufferSize),
+		errCh:     make(chan error, 1),
```

Comment:
Just realized a serious issue with this approach -- if you have both a Writer and a Reader blocked on the `errCh`, and then an error is sent because the connection failed, only *one* of them will be unblocked. We want all readers on `errCh` unblocked (possible reader, writer, as well as the timeout/cancellation wait in inbound).